bike-erp/backend/src/main/services/orderService/BikeOrderService.ts

```diff
@@ -34,10 +35,15 @@ export class BikeOrderService {
                 else if (isNaN(bike.quantity) || bike.quantity < 0) {
                     return rejects({ status: 400, message: "Invalid quantity format, quantity must be a number data type and a positive number" });
                 }
-                    BikeOrderService.getBikeDao().createBike(bike.price, bike.size, bike.color, bike.description, bike.grade, bike.quantity)
-                    .catch(error => {
-                        return rejects(error.message)
-                    })
+                //Posting all the bikes created with their components Id.
+                try {
+                    const response = await BikeOrderService.getBikeDao().createBike(bike.price, bike.size, bike.color, bike.description, bike.grade, bike.quantity);
+                    await BikeOrderService.getBikeDao().linkBikeToComponents(response.bikeId, bike.handle_id, bike.wheel_id, bike.frame_id, bike.seat_id, bike.drive_train_id);
+                    bikeIdList.push(response.bikeId);
```

Comment:
Don't forget to integrate the account_receivable into the order bike process

Reply:
yes, will do that on another branch